Source/ResponseSerialization.swift

```diff
@@ -145,11 +145,13 @@ extension Request {
         - returns: A data response serializer.
     */
     public static func dataResponseSerializer() -> ResponseSerializer<NSData, NSError> {
-        return ResponseSerializer { _, _, data, error in
+        return ResponseSerializer { _, response, data, error in
             guard error == nil else { return .Failure(error!) }
 
-            guard let validData = data where validData.length > 0 else {
-                let failureReason = "Data could not be serialized. Input data was nil or zero length."
+            guard response?.statusCode ?? -1 != 204 else { return .Success(NSData()) }
```

Comment:
I think writing this with parenthesis like this:
```swift
guard (response?.statusCode ?? -1) != 204 else { return .Success(NSData()) }
```
might look better. At a glace I got confused with the precedence of `??` and `!=`.